diff -up Python-2.7.11/Lib/pydoc.py.no_gui Python-2.7.11/Lib/pydoc.py
--- Python-2.7.11/Lib/pydoc.py.no_gui	2015-12-05 19:47:02.000000000 +0000
+++ Python-2.7.11/Lib/pydoc.py	2015-12-13 08:32:00.959919670 +0000
@@ -20,9 +20,6 @@ Run "pydoc -p <port>" to start an HTTP s
 local machine to generate documentation web pages.  Port number 0 can be
 used to get an arbitrary unused port.
 
-For platforms without a command line, "pydoc -g" starts the HTTP server
-and also pops up a little window for controlling it.
-
 Run "pydoc -w <name>" to write out the HTML documentation for a module
 to a file named "<name>.html".
 
@@ -2340,9 +2337,6 @@ def cli():
         writing = 0
 
         for opt, val in opts:
-            if opt == '-g':
-                gui()
-                return
             if opt == '-k':
                 apropos(val)
                 return
@@ -2397,13 +2391,10 @@ def cli():
     Start an HTTP server on the given port on the local machine.  Port
     number 0 can be used to get an arbitrary unused port.
 
-%s -g
-    Pop up a graphical interface for finding and serving documentation.
-
 %s -w <name> ...
     Write out the HTML documentation for a module to a file in the current
     directory.  If <name> contains a '%s', it is treated as a filename; if
     it names a directory, documentation is written for all the contents.
-""" % (cmd, os.sep, cmd, cmd, cmd, cmd, os.sep)
+""" % (cmd, os.sep, cmd, cmd, cmd, os.sep)
 
 if __name__ == '__main__': cli()
