diff -up Python-2.7.11/Lib/test/test_gdb.py.orig Python-2.7.11/Lib/test/test_gdb.py
--- Python-2.7.11/Lib/test/test_gdb.py.orig	2015-12-13 09:33:24.111318172 +0000
+++ Python-2.7.11/Lib/test/test_gdb.py	2015-12-13 09:47:13.797993399 +0000
@@ -216,6 +216,9 @@ class DebuggerTests(unittest.TestCase):
             'Missing separate debuginfo for ',
             'Try: zypper install -C ',
             )
+        ignore_patterns += ('warning: Unable to open',
+                            'Try: yum --disablerepo=',
+                            'Undefined set print command')
         for line in errlines:
             if not line.startswith(ignore_patterns):
                 unexpected_errlines.append(line)
